### PR TITLE
mixin: make range interval configurable in alerts

### LINKED ISSUE #7546: Mimir mixin: alerts make range interval configurable
#### Is your feature request related to a problem? Please describe.

In our case we have scrape interval of 1m, however alerts in the mixin seem to have hardcoded range interval, examples:

* https://github.com/grafana/mimir/blob/main/operations/mimir-mixin/alerts/alerts.libsonnet#L39
* https://github.com/grafana/mimir/blob/main/operations/mimir-mixin/alerts/alertmanager.libsonnet#L24

Because of that some of these alerts won't work in our case.

#### Describe the solution you'd like

Have a config option for the range interval window in alerts, similar to what is already present for recording rules https://github.com/grafana/mimir/blob/main/operations/mimir-mixin/config.libsonnet#L659.

Given the range interval is not the same in the alert the variable could be something like `min_alerts_range_interval` and do something like:
```
1 * %(min_alerts_range_interval)
```
To calculate the desired interval while keeping in mind the minimum possible rate interval.
#### Describe alternatives you've considered

Manually editing the alerts after they are rendered, which kind of defeats the point of mixins imho.

#### Additional context

https://www.robustperception.io/what-range-should-i-use-with-rate/

### PR DESCRIPTION
**Pull request by jmichalek132**
#### Which issue(s) this PR fixes or relates to

Fixes #7546

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: operations/mimir-mixin/config.libsonnet [hunk 1]
```diff
@@ -658,6 +658,12 @@
     // Set to four times the scrape interval to account for edge cases: https://www.robustperception.io/what-range-should-i-use-with-rate/
     recording_rules_range_interval: '1m',
 
+    // Used to calculate range interval in alerts.
+    // In seconds without the `s` at the end, i.e. `30`.
+    // Needed to account for edge cases: https://www.robustperception.io/what-range-should-i-use-with-rate/
+    scrape_interval: 15,
```
Comment (dimitarvdimitrov) [inline operations/mimir-mixin/config.libsonnet:664]: i'd even drop this. My intention with it was to have it more clear what it relates to,  buut we already have `recording_rules_range_interval` which doesn't use that.

File: operations/mimir-mixin/alerts/alertmanager.libsonnet [hunk 2]
```diff
@@ -81,8 +81,8 @@
         {
           alert: $.alertName('AlertmanagerInitialSyncFailed'),
           expr: |||
-            increase(cortex_alertmanager_state_initial_sync_completed_total{outcome="failed"}[1m]) > 0
-          |||,
+            increase(cortex_alertmanager_state_initial_sync_completed_total{outcome="failed"}[%ss]) > 0
+          ||| % $.alertRangeInterval(1),
```
Comment (dimitarvdimitrov) [inline operations/mimir-mixin/alerts/alertmanager.libsonnet:85]: can we move the `s` suffix of the string into `alertRangeInterval` or do we need a numeric output from `alertRangeInterval`?

**Review by dimitarvdimitrov — COMMENTED** [review]
the approach looks good 👍

**Comment by jmichalek132** [comment]
As discussed, the range selector is now dynamic only for alerts with the default value under `10m`.

File: operations/mimir-mixin/alerts/alerts.libsonnet [hunk 3]
```diff
@@ -513,6 +531,7 @@ local utils = import 'mixin-utils/utils.libsonnet';
             kube_statefulset_status_update_revision: groupStatefulSetByRolloutGroup('kube_statefulset_status_update_revision'),
             kube_statefulset_replicas: groupStatefulSetByRolloutGroup('kube_statefulset_replicas'),
             kube_statefulset_status_replicas_updated: groupStatefulSetByRolloutGroup('kube_statefulset_status_replicas_updated'),
+            range_interval: $.alertRangeInterval(15) + ':' + $.alertRangeInterval(1),
```
Comment (dimitarvdimitrov) [inline operations/mimir-mixin/alerts/alerts.libsonnet:534]: we talked privately about keeping anything <=10m hardcoded. Should we keep these hardcoded too? At least the `15m` part. The same applies below

**Review by dimitarvdimitrov — COMMENTED** [review]
thanks for the change. There is one more query I could find in autoscaling.libsonnet which uses a range of 5m. Can you take care of that too? 

https://github.com/grafana/mimir/blob/fd94abe03c44d897fb411732ed9d52d533af69f3/operations/mimir-mixin/alerts/autoscaling.libsonnet#L43

File: operations/mimir-mixin/alerts/alerts.libsonnet (hunk 3 above)
Comment (jmichalek132) [reply operations/mimir-mixin/alerts/alerts.libsonnet]: fixed

**Comment by jmichalek132** [comment]
I also noticed that a lot of recording rules have range interval hardcoded and don't use `recording_rules_range_interval` variable, some of them having `1m` hardcoded https://github.com/grafana/mimir/blob/main/operations/mimir-mixin/recording_rules.libsonnet#L338. I will open a separate PR to address those.

**Comment by jmichalek132** [comment]
Test failure seems to be unrelated to my changes to the mixin.

**Comment by dimitarvdimitrov** [comment]
> Test failure seems to be unrelated to my changes to the mixin.

yep, there's already a filed issue for that test https://github.com/grafana/mimir/issues/5471

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM, thank you :)

**Comment by dimitarvdimitrov** [comment]
would actually be nice to add a changelog entry. My suggestion was something like
```
...
### Mixin 
...
* [ENHANCEMENT] Alerts: allow configuring alerts range interval via `_config.base_alerts_range_interval_minutes`. #7591
```

**Review by dimitarvdimitrov — APPROVED** [review]
lovely, thank you!

**Pull request merged**